### PR TITLE
Purpose templates composition pattern refactor

### PR DESCRIPTION
**Pull request by annalinneajohansson**
<code>Dependant on https://github.com/helsingborg-stad/Modularity/pull/438
</code>
#### Changes
- `add(purpose):` PurposeCollection and PurposeComponent interfaces 
- `refactor(purpose)!:` implement PurposeCollection and PurposeComponent in PurposeFactory to make purpose classes (Event, Place) adhere to the composition pattern
- `refactor(purpose):` Update the Purpose helper methods so that they correspond to the refactored code
- `docs(purpose):` update class and method docblocks to reflect the refactored code

- `refactor(purpose)`: Renaming `purpose` to `content type` throughout
- `refactor(purpose)`: preventing complex types to add other complex types

---

This pull request introduces several key updates aimed at enhancing the functionality of content types (previously called "purposes").

---

_No large language models were hurt in the making of this pull request._

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-865cdv0pq Refaktorering av purpose controllers enl. composition pattern](https://app.clickup.com/t/865cdv0pq)

File: library/Controller/ArchivePurpose.php
Comment (thorbrink) [inline library/Controller/ArchivePurpose.php:1]: This file should probably be renamed to `ArchiveContentType`. Or perhaps both the class and the file should be renamed to something that indicated that this handles archive for places?

**Review by thorbrink — CHANGES_REQUESTED** [review]
Beautiful 🥇 Just added a request for a file/class rename.

File: library/Controller/ArchivePurpose.php
Comment (annalinneajohansson) [reply library/Controller/ArchivePurpose.php:1]: It's meant to generally handle archives containing any kind of content type (not just places), the same way standard Archive.php does in WP so it should be ArchiveContentType

**Pull request merged**